### PR TITLE
Adding farmhash package from google

### PR DESCRIPTION
**Pull request by AP97**
We are trying to build TensorFlow natively as Spack for HackIllinois. This is one of the dependencies for TensorFlow

### CODE REVIEW

File: var/spack/repos/builtin/packages/farmhash/package.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+
+class Farmhash(CMakePackage):
+    """
+    FarmHash provides hash functions for strings and other data.
+    """
+
+    homepage = "https://github.com/google/farmhash"
+
+    version('92e897', git='https://github.com/google/farmhash.git',
```
Comment (davydden) [inline var/spack/repos/builtin/packages/farmhash/package.py]: Please name it `develop`

File: var/spack/repos/builtin/packages/farmhash/package.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    FarmHash provides hash functions for strings and other data.
+    """
+
+    homepage = "https://github.com/google/farmhash"
+
+    version('92e897', git='https://github.com/google/farmhash.git', 
+            commit='92e897b282426729f4724d91a637596c7e2fe28f')
+
+    def patch(self):
+        copyfile(join_path(os.path.dirname(__file__), "CMakeLists.txt"),
```
Comment (davydden) [inline var/spack/repos/builtin/packages/farmhash/package.py]: can you submit your `CMakeList.txt` upstream? I don't think there is any package where we manager our own `CMakeLists.txt`

Comment (tgamblin) [inline var/spack/repos/builtin/packages/farmhash/package.py]: I think we can actually convert this one to an autotools build.  The CMakeLists.txt is from the contributed cmake build in Tensorflow (and lives in its own kind of weird directory there), but it seems like it's not always maintained well.

I'd rather convert this to use the package's native build system (autotools) than keep relying on the external CMakeLists.txt file.

I have he same question as @davydden for the TF cmake maintainers though -- seems like they could submit this upstream, or just use the native build with ExternalPackage_Add.  I am not sure why they don't.

File: var/spack/repos/builtin/packages/farmhash/package.py (hunk 1 above)
Comment (citibeth) [inline var/spack/repos/builtin/packages/farmhash/package.py]: Please name it `0.0.0.1`.

**Pull request merged**